lnbits/extensions/discordbot/crud.py

```diff
@@ -0,0 +1,123 @@
+        VALUES (?, ?, ?, ?, ?, ?)
+        """,
+        (
+            wallet.id,
+            data.admin_id,
+            data.wallet_name,
+            user.id,
+            wallet.adminkey,
+            wallet.inkey,
+        ),
+    )
+
+    user_created = await get_discordbot_user(user.id)
+    assert user_created, "Newly created user couldn't be retrieved"
+    return user_created
+
+
+async def get_discordbot_user(user_id: str) -> Optional[Users]:
+    row = await db.fetchone("SELECT * FROM discordbot.users WHERE id = ?", (user_id,))
+    return Users(**row) if row else None
+
+
+async def get_discordbot_users(user_id: str) -> List[Users]:
+    rows = await db.fetchall(
+        "SELECT * FROM discordbot.users WHERE admin = ?", (user_id,)
+    )
+
+    return [Users(**row) for row in rows]
+
+
+async def delete_discordbot_user(user_id: str) -> None:
+    wallets = await get_discordbot_wallets(user_id)
+    for wallet in wallets:
+        await delete_wallet(user_id=user_id, wallet_id=wallet.id)
+
+    await db.execute("DELETE FROM discordbot.users WHERE id = ?", (user_id,))
+    await db.execute("""DELETE FROM discordbot.wallets WHERE "user" = ?""", (user_id,))
+
+
+### Wallets
+
+
+async def create_discordbot_wallet(
+    user_id: str, wallet_name: str, admin_id: str
+) -> Wallets:
+    wallet = await create_wallet(user_id=user_id, wallet_name=wallet_name)
+    await db.execute(
+        """
+        INSERT INTO discordbot.wallets (id, admin, name, "user", adminkey, inkey)
+        VALUES (?, ?, ?, ?, ?, ?)
+        """,
+        (wallet.id, admin_id, wallet_name, user_id, wallet.adminkey, wallet.inkey),
+    )
+    wallet_created = await get_discordbot_wallet(wallet.id)
+    assert wallet_created, "Newly created wallet couldn't be retrieved"
+    return wallet_created
+
+
+async def get_discordbot_wallet(wallet_id: str) -> Optional[Wallets]:
+    row = await db.fetchone(
+        "SELECT * FROM discordbot.wallets WHERE id = ?", (wallet_id,)
+    )
+    return Wallets(**row) if row else None
+
+
+async def get_discordbot_wallets(admin_id: str) -> Optional[Wallets]:
+    rows = await db.fetchall(
+        "SELECT * FROM discordbot.wallets WHERE admin = ?", (admin_id,)
+    )
+    return [Wallets(**row) for row in rows]
+
+
+async def get_discordbot_users_wallets(user_id: str) -> Optional[Wallets]:
+    rows = await db.fetchall(
+        """SELECT * FROM discordbot.wallets WHERE "user" = ?""", (user_id,)
+    )
+    return [Wallets(**row) for row in rows]
+
+
+async def get_discordbot_wallet_transactions(wallet_id: str) -> Optional[Payment]:
+    return await get_payments(
+        wallet_id=wallet_id, complete=True, pending=False, outgoing=True, incoming=True
+    )
+
+
+async def delete_discordbot_wallet(wallet_id: str, user_id: str) -> None:
+    await delete_wallet(user_id=user_id, wallet_id=wallet_id)
+    await db.execute("DELETE FROM discordbot.wallets WHERE id = ?", (wallet_id,))
```

Comment:
What is the reason that the wallet object is basically copied? Wouldn't it be more efficient to just use the core wallet?

Reply:
hey @fusion44 - stepping in to assist @bitcoingamer21's work. Not sure if you were targeting the `delete_*` part of the code or `crud.py` in general but this extension is basically a clone of [usermanager](https://github.com/lnbits/lnbits-legend/tree/main/lnbits/extensions/usermanager) so its basically creating sub accounts under one master

Reply:
@chrislennon Yes, you're correct. I was targeting the whole crud file. I'll look more into this.